app/src/main/java/com/example/githubusers/screens/main/StartAdapter.kt

```diff
@@ -18,7 +18,7 @@ abstract class ItemUserViewHolder(val binding: ItemUserLayoutBinding): RecyclerV
     )
 }
 
-class StartAdapter(var callBack: (Bundle) -> Unit) : ListAdapter<UsersItemEntity, StartAdapter.ViewHold>(UserListDiffCallBack()) {
+class StartAdapter(val onAdapterClickFun: (Bundle) -> Unit) : PagingDataAdapter<UsersItemEntity, StartAdapter.ViewHold>(UserListDiffCallBack()) {
```

Comment:
better is `onAdapterClick`, fun is explicit and not necessary (you have the type)